### DISCUSSION TITLE
[idea] package.json scripts runner

### DISCUSSION THREAD

**Discussion by escaton** [oxc-project/oxc#4040] [Feature Request]
The ecosystem is currently suffering from a slow invocation of scripts in the `package.json`.
Both Yarn and npm add significant overhead to startup time just because they need to start `node` and load their own code for that.

Here is for example the Yarn's issue, where they are researching the problem: https://github.com/yarnpkg/berry/issues/2575
Tl;DR — there are no easy solutions for yarn. The same goes for npm.

As far as I know, the spot for such a quick standalone script runner is free.
Do you think implementing one fits the Oxc philosophy?

**Comment by Boshen** [comment]
This is out of scope of this project.

Try node v22 `--run` https://nodejs.org/en/blog/announcements/v22-release-announce#running-packagejson-scripts
